src/mbgl/renderer/layers/render_heatmap_layer.cpp

```diff
@@ -55,10 +55,24 @@ void RenderHeatmapLayer::render(PaintParameters& parameters, RenderSource*) {
         const auto size = Size{viewportSize.width / 4, viewportSize.height / 4};
 
         if (!renderTexture || renderTexture->getSize() != size) {
-            const auto textureType = parameters.context.supportsHalfFloatTextures() ?
-                    gl::TextureType::HalfFloat : gl::TextureType::UnsignedByte;
-
-            renderTexture = OffscreenTexture(parameters.context, size, textureType);
+            if (parameters.context.supportsHalfFloatTextures()) {
+                renderTexture = OffscreenTexture(parameters.context, size, gl::TextureType::HalfFloat);
+
+                try {
+                    renderTexture->bind();
+                } catch (const std::runtime_error& ex) {
+                    // can't render to a half-float texture; falling back to unsigned byte one
```

Comment:
Can `context.supportsHalfFlooatTextures` be updated in this case, so we don't need to have this kind of expensive `try..catch` in other parts of the code.

Reply:
Yes! What's the best way to do it? Add a setter like `disableHalfFloatTextures()`, or maybe expose `supportsHalfFloat` as a public property?

Reply:
I don't think a `disableHalFloatTextures()` setter makes sense. It may be better to make the `supprtsHalfFloat` member variable public, similar to whats done in

https://github.com/mapbox/mapbox-gl-native/blob/fcc3808c83102d272006c33fb25f9c0307b466c5/src/mbgl/gl/context.hpp#L291-L298

Reply:
@asheemmamoowala how does https://github.com/mapbox/mapbox-gl-native/pull/11046/commits/3726c5886e338811e504962959e5cfaf59ab204e look?